src/vs/platform/storage/electron-main/storageMain.ts

```diff
@@ -260,7 +261,7 @@ export class GlobalStorageMain extends BaseStorageMain implements IStorageMain {
 
 	get path(): string | undefined {
 		if (!this.options.useInMemoryStorage) {
-			return join(this.environmentService.globalStorageHome.fsPath, GlobalStorageMain.STORAGE_NAME);
+			return join(this.userDataProfilesService.currentProfile.globalStorageHome.fsPath, GlobalStorageMain.STORAGE_NAME);
```

Comment:
As long as storage is not multi-profile aware, this needs to be `defaultProfile` to be safe.